src/buf.rs

```diff
@@ -0,0 +1,98 @@
+use std::path::Path;
+
+use serde::{Deserialize, Serialize};
+
+use crate::error::TonicBufBuildError;
+
+#[derive(Debug, PartialEq, Serialize, Deserialize)]
+pub(crate) struct BufYaml {
+    pub deps: Option<Vec<String>>,
+}
+
+impl BufYaml {
+    pub(crate) fn load(file: &str) -> Result<BufYaml, TonicBufBuildError> {
+        let f = std::fs::File::open(file)
+            .map_err(|e| TonicBufBuildError::new(&format!("failed to read {}", file), e.into()))?;
+
+        let buf: BufYaml = serde_yaml::from_reader(&f)
+            .map_err(|e| TonicBufBuildError::new(&format!("failed to deserialize {}", file), e.into()))?;
+        Ok(buf)
+    }
+}
+
+#[derive(Debug, PartialEq, Serialize, Deserialize)]
```

Comment:
as far as I see, deserialization is not used in this project

Reply:
True, fixed